SSZipArchive/minizip/mz_compat.c

```diff
@@ -871,23 +868,28 @@ int unzOpenCurrentFile3(unzFile file, int *method, int *level, int raw, const ch
     int32_t err = MZ_OK;
     void *stream = NULL;
 
-    if (compat == NULL)
+    if (!compat)
         return UNZ_PARAMERROR;
-    if (method != NULL)
+    if (method)
         *method = 0;
-    if (level != NULL)
+    if (level)
         *level = 0;
 
+    if (mz_zip_entry_is_open(compat->handle) == MZ_OK) {
+        /* zlib minizip does not error out here if close returns errors */
+        unzCloseCurrentFile(file);
+    }
+
```

Comment:
This may be the only change subject to caution.
What do you think of it?

Reply:
Caution is my appropriate feeling, but not enough to block merging in an upgrade to mini zip.